src/transformers/modeling_utils.py

```diff
@@ -1419,9 +1419,8 @@ def _check_and_enable_flash_attn_2(
                 "You are attempting to use Flash Attention 2.0 without specifying a torch dtype. This might lead to unexpected behaviour"
             )
         elif torch_dtype is not None and torch_dtype not in [torch.float16, torch.bfloat16]:
-            raise ValueError(
-                f"Flash Attention 2.0 only supports torch.float16 and torch.bfloat16 dtypes. You passed {torch_dtype}, this might lead to"
-                " unexpected behaviour."
+            logger.warning(
+                "Flash Attention 2.0 only supports torch.float16 and torch.bfloat16 dtypes. Run Training or inference using Automatic Mixed-Precision via `torch.autocast`."
```

Comment:
```suggestion
            logger.warning(
                "Flash Attention 2.0 only supports torch.float16 and torch.bfloat16 dtypes. Not dtype was provided, you should run training or inference using Automatic Mixed-Precision via the `with torch.autocast(device_type="torch_device"):` decorator."
```
a nit to maybe help understand how to use it for beginners